gateway/engine/core/src/main/java/io/apiman/gateway/engine/auth/RequiredAuthType.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2015 JBoss Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.apiman.gateway.engine.auth;
+
+import io.apiman.gateway.engine.beans.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Enum representing Authorization types, with aliases.
+ *
+ * @author Marc Savy <msavy@redhat.com>
+ */
+@SuppressWarnings("nls")
+public enum RequiredAuthType {
+
+    /**
+     * Certificate-based mutual TLS/SSL
+     */
+    MTLS("mtls", "mssl"),
+    /**
+     * BASIC
+     */
+    BASIC("basic"),
+    /**
+     * Default (nothing)
+     */
+    DEFAULT;
+
+    public static final String ENDPOINT_PROPERTIES_KEY = "authType";
```

Comment:
Let's rename this to:

    authorization.type

In the future we can scope the additional authorization endpoint properties this way.  *For example*, basic auth might be:

    authorization.type: basic
    authorization.basic.credentials.username: sadmin
    authorization.basic.credentials.password: abc123!